Remove the double dispatch on Windows for IO

davidfowl opened the pull request:
- The windows IO thread pool was replaced with one that dispatches
to the thread pool. As a result, we remove the extra thread pool dispatch
that we originally had for windows because continuations for IO on windows ran on
the same thread that polls for IO.

See https://github.com/dotnet/runtime/pull/64834 for the thread pool changes.

~PS: I think we need to adjust this for the old non portable threadpool (there's a flag to go back to the old mode that we should probably respect here as well). This should be in .NET 7.~

TBD: Perf tests.

adityamandaleeka commented:
Are we aiming for RC1 for this?

davidfowl commented:
Yes.

adityamandaleeka commented:
OK, can you change the target branch to release/7.0-rc1?

adityamandaleeka commented:
> PS: I think we need to adjust this for the old non portable threadpool (there's a flag to go back to the old mode that we should probably respect here as well).

Is there? :smile: https://github.com/dotnet/runtime/pull/71719

davidfowl commented:
Oh snap its gone!

halter73 reviewed (approved):
Assuming that this improves benchmarks. Or at least doesn't regress it.

davidfowl committed:
Remove the double dispatch on Windows for IO
- The windows IO thread pool was replaced with one that dispatches
to the thread pool. As a result, we remove the extra thread pool dispatch
that we originally had for windows because continuations for IO on windows ran on
the same thread that polls for IO.

dougbu commented:
`servicing-approved` @adityamandaleeka or do we need a Steve email today❔

adityamandaleeka commented:
Waiting for benchmarks that @davidfowl is collecting.

davidfowl commented:
Seems slower

davidfowl commented:
/benchmark plaintext aspnet-citrine-win kestrel

sebastienros commented:
/benchmark plaintext aspnet-citrine-win kestrel

davidfowl commented:
Oh good its faster 😄

davidfowl commented:
/benchmark json aspnet-citrine-win kestrel

adityamandaleeka commented:
Is that 5% RPS real?

davidfowl commented:
/benchmark json plaintext aspnet-citrine-win kestrel

davidfowl commented:
/benchmark json, plaintext aspnet-citrine-win kestrel

davidfowl commented:
/benchmark json,plaintext aspnet-citrine-win kestrel

davidfowl commented:
@adityamandaleeka approve?

JamesNK commented:
The req/s increase is nice. Did you expect a big increase?

davidfowl commented:
> The req/s increase is nice. Did you expect a big increase?

Yes.

dougbu commented:
Approved for RC1 @adityamandaleeka or @Pilchie (if you're back)❔

adityamandaleeka commented:
Approved, this is a good win and getting validation over 2 RCs instead of 1 seems worthwhile.
